lib/DxcSupport/HLSLOptions.cpp

```diff
@@ -213,30 +214,60 @@ StringRefWide::StringRefWide(llvm::StringRef value) {
     m_value = Unicode::UTF8ToWideStringOrThrow(value.data());
 }
 
-static bool GetTargetVersionFromString(llvm::StringRef ref, unsigned *major,
-                                       unsigned *minor) {
-  *major = *minor = -1;
-  unsigned len = ref.size();
-  if (len < 6 || len > 11) // length: ps_6_0 to rootsig_1_0
+// Return true iff Name matches: <2-3 chars> '_' <1 digit> '_' <1-2 digits>
+// On success, *OutMajor = second chunk (single digit), *OutMinor = third chunk
+// (1-2 digits). OutMajor/OutMinor may be null if false is returned
+static bool GetTargetVersionFromString(llvm::StringRef ref, unsigned *OutMajor,
+                                       unsigned *OutMinor) {
+  // Find underscores
+  size_t pos1 = ref.find('_');
+  if (pos1 == llvm::StringRef::npos)
     return false;
-  if (ref[len - 4] != '_' || ref[len - 2] != '_')
+
+  size_t pos2 = ref.find('_', pos1 + 1);
+  if (pos2 == llvm::StringRef::npos)
+    return false;
+
+  // Ensure there are exactly two separators (no extra '_')
+  if (ref.find('_', pos2 + 1) != llvm::StringRef::npos)
     return false;
 
-  char cMajor = ref[len - 3];
-  char cMinor = ref[len - 1];
+  // Construct parts
+  llvm::StringRef Prefix(ref.data(), pos1);
+  llvm::StringRef MajorStr(ref.data() + pos1 + 1, pos2 - (pos1 + 1));
+  llvm::StringRef MinorStr(ref.data() + pos2 + 1, ref.size() - (pos2 + 1));
+
+  // Validate sizes
+  if (Prefix.size() < 2 || Prefix.size() > 3)
+    return false; // first chunk 2..3 chars
```

Comment:
This regresses for root signature targets: `rootsig_1_0`. Since this function is only meant to grab the major and minor versions, I think we should be able to handle this case.